File: src/main/java/com/example/forum/domain/ArticleComment.java
package com.example.forum.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.Objects;

@Getter
@ToString(callSuper = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(indexes = {
        @Index(columnList = "content"),
        @Index(columnList = "createdAt"),
})
@Entity
public class ArticleComment extends AuditingFields {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Setter
    @Column(nullable = false, length = 500)
    private String content;

    @Setter
    @ManyToOne(optional = false)
    private Article article;

    @Setter
    @ManyToOne(optional = false)
    private UserAccount userAccount;

    private ArticleComment(Article article, String content, UserAccount userAccount) {
        this.article = article;
        this.content = content;
        this.userAccount = userAccount;
    }

    public static ArticleComment of(Article article, UserAccount userAccount, String content) {
        return new ArticleComment(article, content, userAccount);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ArticleComment that)) return false;
        return id != null && id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: migration-to-kotlin/src/main/kotlin/com/example/practice/domain/ArticleComment.kt
package com.example.practice.domain

import jakarta.persistence.*

@Table(indexes = [Index(columnList = "content"), Index(columnList = "createdAt")])
@Entity
class ArticleComment(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id : Long = 0L,
    @Column(length = 500, nullable = false) val content : String,

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id")
    val article : Article,

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_account_id")
    val userAccount : UserAccount

) : AuditingFields()